fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Optimizer.java

```diff
@@ -274,6 +276,14 @@ private OptExpression optimizeByCost(ConnectContext connectContext,
             OptimizerTraceUtil.logOptExpression("final plan after physical rewrite:\n%s", finalPlan);
         }
 
+        // collect all mv scan operator
+        collectAllPhysicalOlapScanOperators(result, rootTaskContext);
+        List<PhysicalOlapScanOperator> mvScan = rootTaskContext.getAllPhysicalOlapScanOperators().stream().
+                filter(scan -> scan.getTable().isMaterializedView()).collect(Collectors.toList());
+        // add mv db id to currentSqlDbIds, the resource group could use this to distinguish sql patterns
+        Set<Long> currentSqlDbIds = rootTaskContext.getOptimizerContext().getCurrentSqlDbIds();
```

Comment:
if query's plan has been rewritten, should we exclude the query's dbs? you can refer MVRewriteValidator#validateMV

Reply:
It's not actually possible to remove the original dbs directly, as the mv may be partially rewritten

Reply:
It's possible? if mv replace all scan node?